core/src/main/java/org/elasticsearch/indices/recovery/StartRecoveryRequest.java

```diff
@@ -94,6 +112,11 @@ public void readFrom(StreamInput in) throws IOException {
         targetNode = new DiscoveryNode(in);
         metadataSnapshot = new Store.MetadataSnapshot(in);
         primaryRelocation = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
+            startingSeqNo = in.readZLong();
```

Comment:
Let's standardize on `readZLong/writeZLong`, but let us do that in a separate PR.